dipy/tracking/utils.py

```diff
@@ -1048,3 +1047,47 @@ def min_radius_curvature_from_angle(max_angle, step_size):
         max_angle = np.pi / 2.0
     min_radius_curvature = step_size / 2 / np.sin(max_angle / 2)
     return min_radius_curvature
+
+
+def seeds_directions_pairs(positions, peaks, max_cross=-1):
+    """
+    Pair each seed to the corresponding peaks. If multiple peaks are available
+    the seed is repeated for each.
+
+    Parameters
+    ----------
+    positions : array, (N, 3)
+        Voxel coordinates of the N positions.
+    peaks : array (N, M, 3)
+        Peaks at each position
+    max_cross : int
+        The maximum number of direction to track from each seed in crossing
+        voxels. By default all voxel peaks are used.
+
+    Returns
+    -------
+    seeds : array (K, 3)
+    directions : array (K, 3)
+    """
+
+    if (not len(positions.shape) == 2
+            or not len(peaks.shape) == 3
+            or not positions.shape[0] == peaks.shape[0]
+            or not positions.shape[1] == 3
+            or not peaks.shape[2] == 3
+            or not peaks.shape[1] > 0):
+        raise ValueError("The array shapes of the positions and peaks should"
+                         " be (N,3) and (N,M,3), respectively.")
+
+    seeds = []
+    directions = []
```

Comment:
Why not creating a np.array directly on the correct size. 

this should avoid the slow `append` operation. 

I think you should avoid to work with python list.

Reply:
I don't know the length of the list/array beforehand,  as the number peaks per voxel is variable. Is there a better way to do this with arrays?